File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int profit =0;
        int profit_if_sold_today =0;

        for(int i=0;i<prices.length;i++){
            if(prices[i]<min){
                min = prices[i];
            }

            profit_if_sold_today = prices[i]-min;
            if(profit_if_sold_today>profit){
                profit = profit_if_sold_today;
            }
        }
        return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini=prices[0];
        int cost;
        int profit=0;
        for(int i=1;i<prices.size();i++){
            cost=prices[i]-mini;
            profit=max(profit,cost);
            mini=min(mini,prices[i]);
        }
        return profit;
    }
};